packages/cardano-graphql-services/src/DataProjection/blockHandlers/helpers.ts

```diff
@@ -0,0 +1,79 @@
+import { Asset, BigIntMath } from '@cardano-sdk/core';
+import { AssetMetadata } from '../../MetadataClient/types';
+import { Schema, isAlonzoBlock, isMaryBlock } from '@cardano-ogmios/client';
+
+export const getBlockType = (block: Schema.Block) => {
+  let b: Schema.BlockMary | Schema.BlockAlonzo | undefined;
+  if (isAlonzoBlock(block)) {
+    b = block.alonzo as Schema.BlockAlonzo;
+  } else if (isMaryBlock(block)) {
+    b = block.mary as Schema.BlockMary;
+  }
+  return b;
+};
+
+export const mapMetadata = (serverAssetMetadata: AssetMetadata): Asset.TokenMetadata => ({
+  decimals: serverAssetMetadata.decimals.value,
+  desc: serverAssetMetadata.description?.value,
+  icon: serverAssetMetadata.logo?.value,
+  name: serverAssetMetadata.name?.value,
+  ticker: serverAssetMetadata.ticker?.value,
+  url: serverAssetMetadata.url?.value
+  // Missing fields: sizedIcons
+});
+
+const mapAssets = (assets: { [k: string]: Schema.AssetQuantity } | undefined) =>
+  // Missing: transactionOutput, asset
+  assets ? Object.entries(assets).map(([name, quantity]) => ({ asset: { name }, quantity })) : [];
+
+const mapOutput = (output: Schema.TxOut) =>
+  // Missing: index, transaction, datumHash
```

Comment:
- index: `outputs.indexOf(output)`
- transaction: graph edge to Transaction object
- datumHash: [available in Ogmios schema](https://github.com/CardanoSolutions/ogmios/blob/master/clients/TypeScript/packages/schema/src/index.ts#L1113)